asynciterator.ts

```diff
@@ -309,6 +309,35 @@ export class AsyncIterator<T> extends EventEmitter {
     return '';
   }
 
+  /**
+    Consume all remaining items of the iterator into an array that will be returned asynchronously.
+    @param {object} [options] Settings for array creation
+    @param {integer} [options.limit] The maximum number of items to place in the array.
+   */
+  toArray(options?: { limit?: number }): Promise<T[]> {
+    return new Promise<T[]>((resolve, reject) => {
+      // Collect items in an array
+      const items: T[] = [];
+
+      // Create a named listener so we can easily remove it later
+      const dataListener = (item: T) => {
+        // Resolve earlier if a limit was set and we have reached that limit
+        // @ts-ignore
+        if (items.length >= options?.limit) {
```

Comment:
The `>=` is dangerous, as the `=` part will result in nullish coalescence:

```
[
    -5,    -1,        -0,
     0,   NaN, undefined,
  null, false,     "foo",
   0.1,     1,         3,
     4,     4.5,       6, // only this line should fail
].map(limit => 3 >= limit)
// [
//   true,  true,  true,
//   true,  false, false,
//   true,  true,  false,
//   true,  true,  true,
//   false, false, false
// ]
```

So we really need `!(index < limit)`.

Reply:
I think this is the behaviour we want though, see my comment above: https://github.com/RubenVerborgh/AsyncIterator/pull/40#discussion_r811820332

I handled all the cases in the unit tests (I think), and these should produce the expected results.

Reply:
Right yes, once we are here, we know that there is no zero limit in place.